walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,100 @@
 
 package com.google.sps;
 
+import com.google.sps.Event;
+import com.google.sps.TimeRange;
+import com.google.sps.MeetingRequest;
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+  public Collection<TimeRange> query(Collection<Event> events, 
+                                      MeetingRequest request) {
+    // Get unavailable time for all required attendees in meeting request
+    ArrayList<TimeRange> unavailable = new ArrayList<>();
+    for (String attendee: request.getAttendees()) {
+      ArrayList<TimeRange> meetingTimes = getMeetingTimes(events, attendee);
+      unavailable = merge(unavailable, meetingTimes);
+    }
+
+    Collection<TimeRange> result = new ArrayList<>();
+    for (TimeRange t: getAvailability(unavailable)) {
+      // Get time ranges that are longer than the requested meeting duration
+      if (t.duration() >= request.getDuration()) {
+        result.add(t);
+      }
+    }
+
+    return result;
+  }
+
+  private ArrayList<TimeRange> merge(ArrayList<TimeRange> result, 
+                                      ArrayList<TimeRange> other) {
+    if (result.isEmpty()) {
+      return other;
+    }
+    if (other.isEmpty()) {
+      return result;
+    }
+
+    result.addAll(other);
+    Collections.sort(result, TimeRange.ORDER_BY_START);
+
+    // Merge overlapping ranges
+    for (int i = 0; i < result.size()-1; i++) {
+      TimeRange current = result.get(i);
+      TimeRange next = result.get(i+1);
+
+      if (current.overlaps(next)) {
+        // Case 1:  |_______|
+        //            |__|
+        if (current.contains(next)) {
+          result.remove(next);
+          i--;
```

Comment:
it seems that you don't need to decrease i, since the current one is keep.
Just loop to the next one is okay.